var/spack/repos/builtin/packages/tfel/package.py

```diff
@@ -75,22 +75,63 @@ class Tfel(CMakePackage):
     variant('java', default=False,
             description='Enables java interface')
 
+    variant('build_type', default='Release',
+            description='The build type to build',
+            values=('Debug', 'Release'))
+
     depends_on('java', when='+java')
-    depends_on('python', when='+python')
-    depends_on('python', when='+python_bindings')
+    depends_on('python', when='+python',
+               type=('build', 'link', 'run'))
+    depends_on('python', when='+python_bindings',
+               type=('build', 'link', 'run'))
     depends_on('boost+python', when='+python_bindings')
 
+    extends('python', when='+python')
+
     def cmake_args(self):
 
         args = []
 
-        for i in ['fortran', 'java', 'castem', 'aster',
-                  'abaqus', 'calculix', 'ansys',
-                  'europlexus', 'cyrano', 'lsdyna',
-                  'python', 'python_bindings']:
+        for i in ['fortran', 'java', 'aster', 'abaqus', 'calculix',
+                  'ansys', 'europlexus', 'cyrano', 'lsdyna', 'python']:
             if '+' + i in self.spec:
                 args.append("-Denable-{0}=ON".format(i))
             else:
                 args.append("-Denable-{0}=OFF".format(i))
 
+        if '+castem' in self.spec:
+            args.append("-Dlocal-castem-header=ON")
+        else:
+            args.append("-Dlocal-castem-header=OFF")
+
+        if '+python_bindings' in self.spec:
+            args.append("-Denable-python-bindings=ON")
+        else:
+            args.append("-Denable-python-bindings=OFF")
+
+        if(('+python' in self.spec) or
+           ('+python_bindings' in self.spec)):
+            python = self.spec['python']
+            python_prefix  = python.prefix
+            python_full_version = str(python.version.dotted).split('.')
+            if(int(python_full_version[0]) >= 3):
+                python_version = '{0}.{1}m'.format(python_full_version[0],
+                                                   python_full_version[1])
+            else:
+                python_version = '{0}.{1}'.format(python_full_version[0],
+                                                  python_full_version[1])
+            args.append('-DPYTHON_LIBRARY={0}'.
+                        format(python_prefix.lib +
+                               '/libpython' + str(python_version) + '.so'))
```

Comment:
All of this is rather hacky and won't always work (for example, macOS uses `.dylib` instead of `.so`. The preferred way to do this is to ask the Python package where it's libraries live:
```python
args.append('-DPYTHON_LIBRARY={0}'.format(
    self.spec['python'].libs[0]))
```
See the `libs` attribute of `python` for details on how this is found.